core/nwb.event.yaml

```diff
@@ -0,0 +1,53 @@
+groups:
+- neurodata_type_def: Events
+  neurodata_type_inc: NWBDataInterface
+  doc: A list of timestamps, stored in seconds, of an event
+  attributes:
+  - name: description
+    dtype: text
+    doc: Description of the event.
+  datasets:
+  - name: timestamps
+    dtype: float64
+    dims:
+    - num_times
+    shape:
+    - null
+    doc: Event timestamps, in seconds, relative to the common experiment master-clock stored in 
+      NWBFile.timestamps_reference_time.
+    attributes:
+    - name: unit
+      dtype: text
+      value: seconds
+      doc: Unit of measurement for timestamps, which is fixed to 'seconds'.
+      
+- neurodata_type_def: LabeledEvents
+  neurodata_type_inc: Events
+  doc: A list of timestamps, stored in seconds, of an event that can have different labels. For example,
+    this type could represent the times that reward was given, as well as which of three different 
+    types of reward was given. In this case, the 'labels' dataset would contain values {0, 1, 2},
+    and the 'label_map' would contain three text elements, where the first (index 0) specifies the
+    name of the reward associated with a label = 0, the second (index 1) specifies the name of the
+    reward associated with a label = 1, etc. The labels do not have to start at 0 and do not need to be
+    sequential, e.g. the 'labels' dataset could contain values {0, 10, 100}, and the 'label_map' dataset
+    could contain 101 values, where label_map[0] is 'No reward', label_map[10] is '10% reward', 
+    label_map[100] is 'Full reward', and all other entries in 'label_map' are the empty string.
+  datasets:
+  - name: labels
```

Comment:
How would you feel about making this dataset optional? If it is not supplied, the label keys are `0, 1, 2, ...` Another option would be to keep this as is and build that into the API.

Reply:
Might be a good use of DynamicTable. Users could then add additional metadata to their labels.